.github/workflows/ci_msbuild.yaml

```
@@ -0,0 +1,110 @@
+# SPDX-FileCopyrightText: 2023, Yaskawa America, Inc.
+# SPDX-FileCopyrightText: 2023, Delft University of Technology
+#
+# SPDX-License-Identifier: CC0-1.0
+
+name: "CI: build VS proj"
+on:
+  pull_request:
+    branches:
+      - main
+  workflow_dispatch:
+
+jobs:
+  msbuild:
+    runs-on: windows-2022
+
+    strategy:
+      # keep jobs running even if one fails (we want to know on which
+      # controllers and for which ROS 2 versions builds fail / succeed)
+      fail-fast: false
+      matrix:
+        # controller: [yrc1000, yrc1000u]
+        # ros2_codename: [humble, galactic, foxy]
+        controller: [yrc1000]
+        ros2_codename: [humble]
```

We could extend this to include all the controllers we currently support, for all the ROS 2 versions we currently support.

```suggestion
        controller: [yrc1000, yrc1000u]
        ros2_codename: [humble, galactic, foxy]
```